src/App.tsx

```diff
@@ -1,24 +1,217 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import {
+  createTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+import { Header } from './components/header';
+import { TodoList } from './components/todos';
+import { Error } from './components/error';
+import { Filter } from './components/filter';
+import { FilterOptions } from './types/FilterOptions';
+import { UpdateTodoArgs } from './types/UpdateTodoArgs';
 
-const USER_ID = 0;
+const USER_ID = 10995;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [
+    filterMethod,
+    setFilterMethod,
+  ] = useState<FilterOptions>(FilterOptions.ALL);
+  const [isHidden, setIsHidden] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => setError('Error: cannot upload todos'));
+  }, []);
+
+  useEffect(() => {
+    let errorTimer: number;
+    let hiddenTime: number;
+
+    if (error) {
+      hiddenTime = window.setTimeout(() => {
+        setIsHidden(true);
+      }, 2000);
+      errorTimer = window.setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => {
+      clearTimeout(errorTimer);
+      clearTimeout(hiddenTime);
+    };
+  }, [error]);
+
+  const completedTodos = todos.filter((todo) => todo.completed);
+  const activeTodos = todos.filter((todo) => !todo.completed);
```

Comment:
Would be great to use useMemo() here to improve performance by caching the filtered results and avoiding redundant filtering when the todos array remains unchanged between renders.